src/main/java/com/melon/portfoliomanager/services/HttpService.java

```diff
@@ -0,0 +1,61 @@
+package com.melon.portfoliomanager.services;
+
+import com.melon.portfoliomanager.dtos.responses.StockPricesDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.net.URI;
+
+@Service
+public class HttpService {
```

Comment:
Totally agree - I started writing this service exactly with the idea described in your second sentence in your comment - to be something like an abstraction over sending http requests in order to hide all of the technical details of using the ```webClient``` http client that we use. Then I thought it will be unnecessary to do that at this point of the creation of our application since there are no expectations to actively use the ```webClient``` in future. So I decided to write ad-hoc method named ```getStockPrices```which will do only what it is saying it will do. So may be I should just choose better name for the service in order not to mislead the reader of the code.
As for the client vs. service terminology - I can only partially agree with what you say since because there is some custom error handling logic inside the ```HttpService.getStockPrices()``` method which according to me qualifies it more as a service rather than a client, since a client will only send the request and return a response with no error handling logic (the error handling logic will be written by the user of the client).